### ISSUE TITLE
Gendoc only takes the first related list in its foreach loop

### ISSUE THREAD

**Issue by Luke1982** [tsolucio/corebos#684]
https://github.com/tsolucio/corebos/blob/bebbf70e9967ea6b63960915b1323e15cb77069a/modules/evvtgendoc/compile.php#L702

This may be an edge-case, but I know of multiple cases where two modules were related through multiple lists (a UI10 and M:M for example). Gendoc only takes the first list. May be good to use them all.

**Comment by joebordes** [comment]
We did not consider this case when creating the GenDoc extension.
The idea of mixing the records of all the related lists doesn't seem right to me as each one may have different "meaning". I think the correct approach would be to permit indicating which related list you want to work with in the foreach and maybe even have an option to say "mix them all together".

In any case, this is not a trivial task, at least it doesn't sound easy nor backward compatible.

We'll keep it in mind but I doubt we will implement it.

**Comment by Luke1982** [comment]
Yes permitting the name of the related list to indicate a certain list would seem like a good solution to me

**Comment by Luke1982** [comment]
> We did not consider this case when creating the GenDoc extension.
> The idea of mixing the records of all the related lists doesn't seem right to me as each one may have different "meaning". I think the correct approach would be to permit indicating which related list you want to work with in the foreach and maybe even have an option to say "mix them all together".
> 
> In any case, this is not a trivial task, at least it doesn't sound easy nor backward compatible.
> 
> We'll keep it in mind but I doubt we will implement it.

I disagree here. There are plenty of use cases where you would create different related lists for the same module that all have different meaning. The "meaning" would be filtered by setting conditions in the foreach loop, not by just discarding loops.
